internal/store/firestore.go

```diff
@@ -0,0 +1,154 @@
+package firestore
+
+import (
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/firestore"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"pantheon.io/edrt-policy-docs-functions/internal/types"
+)
+
+const hostnameCollectionName = "hostnames"
+const hostnameMetadataCollectionName = "hostnameMetadata"
+const edgeLogicCollectionName = "edgelogic"
+
+type Firestore struct {
+	logger    *logrus.Logger
+	projectId string
+}
+
+func New(logger *logrus.Logger, projectId string) Firestore {
+	return Firestore{
+		logger:    logger,
+		projectId: projectId,
+	}
+}
+
+func (f *Firestore) Read(ctx context.Context) interface{} {
+	client, err := firestore.NewClient(ctx, f.projectId)
+	if err != nil {
+		f.logger.Fatalf("Unable to create firestore client %s", err)
+	}
+
+	collection := client.Collection("foo")
+	doc := collection.Doc("bar")
+	data, err := doc.Get(ctx)
+	if err != nil {
+		f.logger.Fatalf("Unable to get document %s", err)
+	}
+
+	theData := data.Data()["baz"]
```

Comment:
You can reuse the same approach you use below: `DataTo` instead of `Data`

Reply:
Similar to my comment above, this function is going away and yes, DataTo is a better option here